mafContrib/transientAsciiSEDMetric.py

```diff
@@ -0,0 +1,662 @@
+            tranisent SED.
+        num_pre_time : int, optional
+            Number of observations (in any filter(s)) to demand before pre_time,
+            before saying a transient has been detected.
+            Default 0.
+        pre_time : float, optional
+            The time by which num_pre_time detections are required (in days).
+            Default 5.0.
+        num_filters : int, optional
+            Number of filters that need to be observed for an object to be
+            counted as detected. Default 1. (if num_per_lightcurve is 0, then
+            this will be reset to 0).
+        filter_time : float, optional
+            The time within which observations in at least num_filters are
+            required (in days). Default None (no time constraint).
+        num_per_lightcurve : int, optional
+            Number of sections of the light curve that must be sampled above
+            the detect_SNR theshold for the light curve to be counted.
+            For example, num_per_lightcurve = 2 means a light curve is only
+            considered detected if there is at least 1 observation in the first
+            half of the LC, and at least one in the second half of the LC.
+            num_per_lightcurve = 4 means each quarter of the light curve must
+            be detected to count. Default 1.
+        num_phases_to_run : int, optional
+            Sets the number of phases that should be checked.
+            One can imagine pathological cadences where many objects pass the
+            detection criteria, but would not if the observations were offset
+            by a phase-shift. Default 1.
+        output_data : bool, optional
+            If True, metric returns full lightcurve at each point. Note that
+            this will potentially create a very large metric output data file.
+            If False, metric returns the number of transients detected.
+    """
+
+    def __init__(
+        self,
+        ascii_file,
+        metric_name="TransientAsciiSEDMetric",
+        mjdCol="expMJD",
+        m5Col="fiveSigmaDepth",
+        filterCol="filter",
+        survey_duration=10.0,
+        survey_start=None,
+        detect_SNR={"u": 5, "g": 5, "r": 5, "i": 5, "z": 5, "y": 5},
+        z=0.1,
+        num_pre_time=0,
+        pre_time=5.0,
+        num_filters=1,
+        filter_time=None,
+        num_per_lightcurve=1,
+        num_phases_to_run=1,
+        output_data=False,
+        **kwargs
+    ):
+        # Set all initial attributes of the metric.
+        self.mjdCol = mjdCol
+        self.m5Col = m5Col
+        self.filterCol = filterCol
+        self.output_data = output_data
+        self.z = z
+
+        # If it is specified that the lightcurves should be returned rather
+        # than the summary statistics, this changes the output of the metric,
+        # and thus needs an alternate definition of the output type and units
+        # when calling the init of the base class.
+        if self.output_data:
```

Comment:
But can you just define those units and metricDtype in an if block, and then pass the variable to the super method? Having two different calls to super is confusing (kind of like having two different points where you can return from a function) and can lead to maintenance issues down the line.